src/docs/development/add-to-app/android/add-flutter-screen.md

```diff
@@ -205,30 +132,35 @@ public class MyApplication extends Application {
   Flutter content.
 {{site.alert.end}}
 
-With a pre-warmed `FlutterEngine`, you now need to instruct your
-`FlutterActivity` to use the pre-warmed `FlutterEngine` instead of creating a
-new one. To accomplish this, subclass `FlutterActivity` and override the
-`provideFlutterEngine()` method:
+With a pre-warmed, cached `FlutterEngine`, you now need to instruct your
+`FlutterActivity` to use the cached `FlutterEngine` instead of creating a
+new one. To accomplish this, use `FlutterActivity`'s `withCachedEngine()`
+builder:
 
 ```java
-public class MyFlutterActivity extends FlutterActivity {
+myButton.addOnClickListener(new OnClickListener() {
   @Override
-  @NonNull
-  protected FlutterEngine provideFlutterEngine(@NonNull Context context) {
-    return ((MyApplication) context.getApplicationContext()).getFlutterEngine();
+  public void onClick(View v) {
+    startActivity(
+      FlutterActivity
+        .withCachedEngine("my_engine_id")
+        .build(currentActivity)
+      );
   }
-}
+});
 ```
 
-Now, when you launch `MyFlutterActivity`, there is significantly less delay in
+When using the `withCachedEngine()` factory method, pass the same ID that you
+used when caching the desired `FlutterEngine`.
+
+Now, when you launch `FlutterActivity`, there is significantly less delay in
 the display of Flutter content.
 
 {{site.alert.note}}
-  When re-using a cached engine, you assume responsibility for
-  showing the desired content for the given `Activity` that's using the
-  `FlutterEngine`. Strategies for switching content in a shared `FlutterEngine`
-  are beyond the scope of this guide. TODO(mattcarroll): link to
-  resource.
+  When re-using a cached engine across multiple `FlutterActivity` instances, you 
+  assume responsibility for showing the desired content for the given `Activity`.
```

Comment:
Not sure I understood this section. Perhaps give an example of such a responsibility